modules/ROOT/pages/design-common-errors-08.adoc

```diff
@@ -0,0 +1,360 @@
+    }
+  ]
+}
+----
+
+== Common Error 3
+// APIMF-852 and 853
+
+
+*_Using a non-integer data type for the example of a property that is an integer_*
+
+The data type for the example of a property that is an integer must be an integer. As an example of the error, this snippet shows that the data type of the example does not match the data type defined for `unit_price`
+
+
+----
+#%RAML 0.8
+title: ExampleRAML
+
+/products:
+  /product/{product_id}:
+    description: Get product instance details
+    get:
+      responses:
+        200:
+          body:
+            application/json:
+              schema: |
+                {
+                  "$schema": "http://json-schema.org/draft-03/schema",
+                  "id": "http://jsonschema.net",
+                  "properties": {
+                      "unit_price": {
+                          "type": "integer"
+                      }
+                  }
+                }
+              example: |
+                {
+                  "unit_price": 19.00
+                }
+----
+
+As another example of the error, the data type of the example does not match the data type defined for `uriParameters`.
+
+
+----
+#%RAML 0.8
+title: Order Entry API
+/orders:
+  /{id}:
+    uriParameters:
+      id:
+        type: integer
+        example: "451"
+----
+
+== Common Error 4
+// APIMF-896
+
+*_Not using a required property in the example of a schema that defines that property_*
+
+An example of a schema that defines a required property must use that property. In this snippet, the property that is defined in the second `id` attribute is misspelled: `sfOrderId` should be `sOrderId`. The second snippet shows the example JSON response in which `sOrderId` is defined. The example, therefore, is not in fact using the required property because the property was not defined correctly.
+
+
+----
+#%RAML 0.8
+title: ExampleRAML
+/order:
+  /{id}:
+    get:
+      responses:
+        200:
+          body:
+            application/json:
+              schema: |
+                {
+                  "type":"object",
+                  "$schema": "http://json-schema.org/draft-03/schema",
+                  "id": "http://com.mulesoft.demo.orders.get.json.order",
+                  "properties":{
+                      "property": {
+                        "type":"string",
+                        "id": "http://com.mulesoft.demo.orders.create.json.get.sfOrderId",
+                        "required":true
+                      }
+                  }
+                }
+              example: |
+                {
+                  "anotherProp": 14523
+                }
+----
+
+
+----
+{
+  "orderId": 14523,
+  "sOrderId": "fadfead3524523",
+  "sfAccountId": "fedfes3653635",
+  "orderName": "Order From Someone",
+  "total": 174.92,
+  "orderType": "E-Commerce Order",
+  "description": "A good order description",
+  "orderDate": "04-03-1980"
+}
+----
```

Comment:
I don't know were you use this example at the previous raml api.

Reply:
I'm sorry, but I don't understand your comment. Would you please rephrase it?

Reply:
I've misunderstood the ticket and have made changes to the write-up here. Would you have a look at Common Error 4 again?